src/components/translation-builder/translation-builder.component.tsx

```diff
@@ -124,7 +139,28 @@ const TranslationBuilder: React.FC<TranslationBuilderProps> = ({ formSchema, onU
             }}
           />
 
-          <IconButton kind="ghost" label={t('downloadTranslation', 'Download translation')} size="md">
+          <IconButton
+            kind="ghost"
+            label={t('downloadTranslation', 'Download translation')}
+            size="md"
+            onClick={() => {
+              setDownloadError(null);
+              if (!downloadableTranslationResource) {
+                if (langCode !== 'en') {
+                  setDownloadError(t('noTranslationForLang', 'No translations found for selected language.'));
+                }
+                return;
+              }
+              const url = URL.createObjectURL(downloadableTranslationResource);
+              const link = document.createElement('a');
+              link.href = url;
+              link.download = `${formSchema?.name}_translations_${langCode}.json`;
+              document.body.appendChild(link);
+              link.click();
+              document.body.removeChild(link);
+              URL.revokeObjectURL(url);
+            }}
```

Comment:
This should have been added to the translation file, please commit those changes too